### PR TITLE
Add support to use imagePullSecret on pod creation

### PR DESCRIPTION
**Pull request by leslau**
Make it possible to use imagePullSecret to pull zalenium images from private repositories.

Pods created by zalenium will use the same imagePullSecret.

### CODE REVIEW

File: charts/zalenium/templates/_pod-template.yaml [hunk 1]
```diff
@@ -9,6 +9,10 @@ spec:
     fsGroup: {{ .Values.hub.securityContext.fsGroup }}
     runAsUser: {{ .Values.hub.securityContext.runAsUser }}
   {{- end }}
+  {{- if .Values.hub.pullSecrets }}
+  imagePullSecrets:
+    - name: {{ .Values.hub.pullSecrets }}
```
Comment (arnaud-deprez) [inline charts/zalenium/templates/_pod-template.yaml]: Please rename `pullSecrets` to `imagePullSecret` to follow kubernetes pod template semantic and as you set only one secret.

File: charts/zalenium/values.yaml [hunk 2]
```diff
@@ -11,6 +11,10 @@ hub:
   ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
   pullPolicy: "IfNotPresent"
 
+  ## Specify secrets to pull images from private repositories
+  ## ref: https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
+  #pullSecrets: "secret1"
```
Comment (arnaud-deprez) [inline charts/zalenium/values.yaml]: Same: imagePullSecret

File: src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java [hunk 3]
```diff
@@ -102,6 +103,7 @@ public KubernetesContainerClient(Environment environment,
             discoverHostAliases();
             discoverNodeSelector();
             discoverTolerations();
+            discoverPullSecrets();
```
Comment (arnaud-deprez) [inline src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java]: I think it's better to keep the full name as well. Please rename to `discoverImagePullSecrets()`

File: src/test/java/de/zalando/ep/zalenium/container/kubernetes/PodConfigurationTest.java [hunk 4]
```diff
@@ -128,4 +128,14 @@ public void setTolerations() {
         assertThat(podConfiguration.getTolerations().size(), is(1));
         assertThat(podConfiguration.getTolerations().get(0), is(toleration));
     }
+
+    @Test
+    public void setPullSecrets() {
```
Comment (arnaud-deprez) [inline src/test/java/de/zalando/ep/zalenium/container/kubernetes/PodConfigurationTest.java]: I would prefer to rename it as `testSetImagePullSecrets` as well

**Comment by diemol** [comment]
Thanks for your contribution @leslau and @arnaud-deprez for the review!

**Pull request merged**